pkg/agent/controller/egress/local_ip_detector.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package egress
+
+import (
+	"net"
+)
+
+type LocalIPDetector interface {
+	IsLocalIP(ip string) (bool, error)
+}
+
+type localIPDetector struct{}
+
+// IsLocalIP checks if the provided IP is configured on the Node.
+// TODO: Instead of listing all IP addresses every time, it can maintain a cache and subscribe the IP address change to
```

Comment:
We need notification to controller for changes too, so controller can handle IP add/delete and update DP correctly.

Reply:
I found golang API to watch ip address, will update this interface soon.